File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    var ans = 0;
    var cumulative_sum = 0;
    d.sort((a, b) => a - b);
    for (num of d) {
        ans += 1;
        cumulative_sum += num;
        if (cumulative_sum >= budget) {
            if (cumulative_sum > budget)
                ans -= 1;
            break;
        }
    }
    return ans;
}

File: 프로그래머스/1/12982. 예산/예산.py
def solution(d, budget):
    d.sort()  # 신청 금액별 오름차순 정렬
    
    now, total_money = 0, 0
    for money in d:  # 모든 부서에 지원 가능한 경우
        # 지원금이 예산을 넘어가기 전에 종료
        if total_money + money > budget:
            break
        total_money += money
        now += 1
    
    return now